packages/salesforcedx-vscode-core/src/commands/forceRenameLightningComponent.ts

```diff
@@ -0,0 +1,129 @@
+const REANME_INPUT_DUP_ERROR = 'rename_comp_input_dup_error';
+const RENAME_COMP_WARNING = 'rename_comp_warning';
+
+export class RenameLwcComponentExecutor extends LibraryCommandletExecutor<string> {
+  private sourceFsPath: string;
+  private responseText: string | undefined;
+  constructor(sourceFsPath: string, responseText: string | undefined) {
+    super(
+      nls.localize(RENAME_LIGHTNING_COMPONENT_EXECUTOR),
+      RENAME_LIGHTNING_COMPONENT_EXECUTOR,
+      OUTPUT_CHANNEL
+    );
+    this.sourceFsPath = sourceFsPath;
+    this.responseText = responseText;
+  }
+
+  public async run(
+    response: ContinueResponse<string>,
+    progress?: vscode.Progress<{ message?: string | undefined; increment?: number | undefined; }>,
+    token?: vscode.CancellationToken
+    ): Promise<boolean> {
+      if (this.sourceFsPath) {
+        if (this.responseText) {
+          renameComponent(this.sourceFsPath, this.responseText);
+          return true;
+        }
+      }
+      return false;
+  }
+}
+
+export async function forceRenameLightningComponent(sourceUri: vscode.Uri) {
+  const sourceFsPath = sourceUri.fsPath;
+  const inputOptions = {
+    placeHolder: nls.localize(RENAME_INPUT_PLACEHOLDER),
+    promopt: nls.localize(REAME_INPUT_PROMPT)
+  } as vscode.InputBoxOptions;
+
+  const responseText = await vscode.window.showInputBox(inputOptions);
+  if (sourceFsPath) {
+    const commandlet = new SfdxCommandlet(
+      new SfdxWorkspaceChecker(),
+      new FilePathGatherer(sourceUri),
+      new RenameLwcComponentExecutor(sourceFsPath, responseText)
+    );
+    await commandlet.run();
+  }
+}
+
+function renameComponent(sourceFsPath: string, newName: string) {
+  const componentPath = getComponentPath(sourceFsPath);
+  const componentName = path.basename(componentPath);
+  checkForDuplicateName(componentPath, newName);
+  const items = fs.readdirSync(componentPath);
+  if (items) {
+    for (const item of items) {
+      // only rename the file that has same name with component
+      if (isNameMatch(item, componentName, componentPath)) {
+        const newItem = item.replace(componentName, newName);
+        fs.renameSync(
+          path.join(componentPath, item),
+          path.join(componentPath, newItem));
+      }
+    }
+  }
+  const newComponentPath = path.join(path.dirname(componentPath), newName);
+  fs.renameSync(
+    componentPath,
+    newComponentPath);
+  notificationService.showWarningMessage(nls.localize(RENAME_COMP_WARNING));
+}
+
+function getComponentPath(sourceFsPath: string): string {
+  const stats = fs.statSync(sourceFsPath);
+  return stats.isFile() ? path.dirname(sourceFsPath) : sourceFsPath;
+}
+
+function checkForDuplicateName(componentPath: string, newName: string) {
+  if (isDuplicate(componentPath, newName)) {
+    notificationService.showErrorMessage(nls.localize(REANME_INPUT_DUP_ERROR));
+    throw new Error(format(nls.localize(REANME_INPUT_DUP_ERROR)));
```

Comment:
`nls.localize(REANME_INPUT_DUP_ERROR)` is called twice here.  How about something like:
```
errorMessage = nls.localize(REANME_INPUT_DUP_ERROR)
```
...and then use `errorMessage`?